expression/util.go

```diff
@@ -526,3 +527,57 @@ func GetParamExpression(ctx sessionctx.Context, v *driver.ParamMarkerExpr, useCa
 	}
 	return value, nil
 }
+
+// ParamToByItemNode generate ByItem node from ParamMarkerExpr.
+func ParamToByItemNode(ctx sessionctx.Context, v *driver.ParamMarkerExpr) (*ast.ByItem, bool, error) {
+	value, err := GetParamExpression(ctx, v)
+	if err != nil {
+		return nil, true, errors.Trace(err)
+	}
+	str, isNull, err := GetStringFromConstant(ctx, value)
+	if err != nil {
+		return nil, true, errors.Trace(err)
+	}
+	if isNull {
+		return nil, true, nil
+	}
+	pos, err := strconv.Atoi(str)
```

Comment:
```go
if err != nil {
	// handle err
}
// construct new ByItem and return
```

Reply:
The updated PR should tiny up the codes.